src/components/TeacherComponents/OnboardingForm/OnboardingForm.schema.ts

```diff
@@ -18,17 +18,31 @@ export type RoleSelectFormProps = RoleSelectFormValues & {
   onSubmit: (values: RoleSelectFormValues) => Promise<void>;
 };
 
-export const schoolSelectFormSchema = z.object({
-  ...baseSchema.shape,
-  school: z
-    .string({
-      errorMap: () => ({
-        message: "Select school",
-      }),
-    })
-    .min(1, "Select school"),
-  schoolName: z.string().optional(),
-});
+export const schoolSelectFormSchema = z
+  .object({
+    school: z
+      .string({
+        errorMap: () => ({
+          message: "Select school",
+        }),
+      })
+      .min(1, "Select school"),
+    schoolName: z.string().optional(),
+    ...baseSchema.shape,
+  })
+  .or(
```

Comment:
ooh nice I didn't know you could use `or` in a schema like this 🤌 is it possible to share the baseSchema between them so we don't need to duplicate it?

Reply:
I think they're technically two different objects so needs to be twice